pkg/bootstrap/init.go

```diff
@@ -251,51 +200,33 @@ func yamlToObject(yamlContent []byte) (*unstructured.Unstructured, error) {
 	return obj, nil
 }
 
-func createOrUpdateResourcesFromDir(dir string, client client.Client, reader client.Reader) error {
-	files, err := ioutil.ReadDir(dir)
+func createOrUpdateResourcesFromAnnotation(client client.Client, reader client.Reader) error {
+	deploy, err := getDeployment(reader)
 	if err != nil {
 		return err
 	}
-
-	var yamlFiles []string
-
-	for _, file := range files {
-		if filepath.Ext(file.Name()) == ".yaml" {
-			yamlFiles = append(yamlFiles, file.Name())
-		}
-	}
-
-	for _, file := range yamlFiles {
-		yamlContent, err := ioutil.ReadFile(filepath.Join(dir, file))
-		if err != nil {
-			return err
-		}
-		if err := createOrUpdateFromYaml(yamlContent, client, reader); err != nil {
-			return err
+	annotations := deploy.Spec.Template.GetAnnotations()
+	for _, res := range CsResources {
+		if res == "extraResources" {
+			klog.Info("create extra resource")
+			extResources := strings.Split(annotations[res], ",")
+			for _, extRes := range extResources {
+				extRes = strings.TrimSpace(extRes)
+				if er, ok := annotations[extRes]; ok {
+					klog.Infof("create extra resource: %s", extRes)
+					if err := createOrUpdateFromYaml([]byte(er), client, reader); err != nil {
+						return err
+					}
+				}
+			}
+		} else {
+			if r, ok := annotations[res]; ok {
+				klog.Infof("create resource: %s", res)
+				if err := createOrUpdateFromYaml([]byte(r), client, reader); err != nil {
+					return err
+				}
+			}
 		}
 	}
-
-	return nil
-}
-
-func deleteExistingODLM(client client.Client) error {
```

Comment:
Should we delete this section?

Reply:
@DanielXLee We still need to keep this part as we may release 3.4.1 to stable-v1 channel.

Reply:
this func not reference, so I delete it.

Reply:
It was referred in https://github.com/IBM/ibm-common-service-operator/blob/master/pkg/bootstrap/init.go#L67

Reply:
got it, back